### ISSUE TITLE
[question] resource reservation does not consider scale up via cluster-autoscaler

### ISSUE THREAD

**Issue by lukasmrtvy** [koordinator-sh/koordinator#2113]
Hi, I don't understand correctly the reservation concept. There is an alternative mentioned in docs ( https://koordinator.sh/docs/designs/resource-reservation ) with balloon/pause pods, but these are triggering the scale-up event, thus it's possible to overprovision the cluster. Reservation does not work like this.  I would like to have additional reservation for ad-hoc pods ( lets say to have two "spare" nodes to cover some peaks ). Is it possible? Thanks

**Comment by saintube** [comment]
@lukasmrtvy Hi, the reservation would not trigger the scale-up while the vanilla version of the cluster autoscaler is not aware of the reservation's scheduling result. For this purpose, we need to develop the CA to cooperate with the current version of the reservation. It is a good point for us to support cluster overprovision with the vanilla CA.
However, the reservation can reserve the allocatable resources such that other unmatched pods cannot preempt them, which frees us from allocating the reserved resources by the preemption.
/cc @ZiMengSheng

**Comment by lukasmrtvy** [comment]
Kueue has implemented https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/provisioning-request.md,  could it be perhaps similar for Koordinator ?

**Comment by saintube** [comment]
> Kueue has implemented https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/provisioning-request.md, could it be perhaps similar for Koordinator ?

@lukasmrtvy Yes, they are similar in reserving resources for owner pods. I appreciate your reminder. We are interested in this protocol and maybe we can resolve the above problem with this.

**Comment by lukasmrtvy** [comment]
@saintube Thanks. The lack of support for CA is the main problem preventing me from adopting Koordinator, so if it is going to be implemented anytime soon, then it would be superb.

**Comment by JBinin** [comment]
Kindly to ask if there is somebody working for this issue? If not, I'm glad to undertake it.

**Comment by saintube** [comment]
> Kindly to ask if there is somebody working for this issue? If not, I'm glad to undertake it. @saintube

@JBinin Yes. Welcome to contribute!

**Comment by lukasmrtvy** [comment]
any update here?

**Comment by ZiMengSheng** [comment]
Closing this stale issue — the assignee has been inactive for ~2 years and the proposal hasn't moved since. The concern itself (reservations not accounting for nodes added by cluster-autoscaler) is a real one; if someone wants to pick it up, please open a fresh issue or comment here and we'll reopen. The Reservation implementation has evolved substantially since this was filed, so a fresh discussion would start from the current code anyway.
